extensions/cl_mobileye_reservation_sets.asciidoc

```diff
@@ -0,0 +1,545 @@
+    synchronization-point objects in _sync_point_wait_list_ are not valid
+    synchronization-points.
+  * `CL_OUT_OF_RESOURCES` if there is a failure to allocate resources
+    required by the OpenCL implementation on the device.
+  * `CL_OUT_OF_HOST_MEMORY` if there is a failure to allocate resources
+    required by the OpenCL implementation on the host.
+--
+
+[open,refpage='clRetainReservationSetMOBILEYE',desc='Increments the reservation-set object reference count.',type='protos']
+--
+To retain a reservation-set object, call the function
+
+include::{generated}/api/protos/clRetainReservationSetMOBILEYE.txt[]
+include::{generated}/api/version-notes/clRetainReservationSetMOBILEYE.asciidoc[]
+
+  * _reservation_set_ specifies the reservation-set object to be retained.
+
+The _reservation_set_ reference count is incremented.
+
+*clRetainReservationSetMOBILEYE* returns `CL_SUCCESS` if the function is
+executed successfully.
+Otherwise, it returns one of the following errors:
+
+  * `CL_INVALID_COMMAND_BUFFER_KHR` if _command_buffer_ is not a valid command
+    -buffer.
+  * `CL_OUT_OF_RESOURCES` if there is a failure to allocate resources required
+    by the OpenCL implementation on the device.
+  * `CL_OUT_OF_HOST_MEMORY` if there is a failure to allocate resources
+    required by the OpenCL implementation on the host.
+--
+
+[open,refpage='clReleaseReservationSetMOBILEYE',desc='Decrements the reservation-set object reference count.',type='protos']
+--
+To release a reservation-set object, call the function
+
+include::{generated}/api/protos/clReleaseReservationSetMOBILEYE.txt[]
+include::{generated}/api/version-notes/clReleaseReservationSetMOBILEYE.asciidoc[]
+
+  * _reservation_set_ specifies the reservation-set object to be released.
+
+The _reservation_set_ reference count is decremented.
+
+*clReleaseReservationSetMOBILEYE* returns `CL_SUCCESS` if the function is
+executed successfully.
+Otherwise, it returns one of the following errors:
+
+  * `CL_INVALID_COMMAND_BUFFER_KHR` if _reservation_set_ is not a valid
+   reservation-set object.
+  * `CL_OUT_OF_RESOURCES` if there is a failure to allocate resources required
+    by the OpenCL implementation on the device.
+  * `CL_OUT_OF_HOST_MEMORY` if there is a failure to allocate resources
+    required by the OpenCL implementation on the host.
+
+After the _reservation_set_ reference count becomes zero and the command-buffer
+associated with _reservation_set_ is not executing, the reservation-set object
+is deleted.
+--
+
+[open,refpage='clCommandNDRangeKernelReservationSetMOBILEYE',desc='Record an ND-range kernel command into a reservation-set.',type='protos']
+--
+To record an ND-range kernel command into a reservation-set, call the function:
+include::{generated}/api/protos/clCommandNDRangeKernelReservationSetMOBILEYE.txt[]
+include::{generated}/api/version-notes/clCommandNDRangeKernelReservationSetMOBILEYE.asciidoc[]
```

Comment:
Just a though, but rather than defining a new entry-point you could define a new `cl_command_properties_khr` property for `clCommandNDRangeKernelKHR` that lets the user set the reservation set. Then rather than having to copy all the spec language for `clCommandNDRangeKernelKHR`  and keep it up to date with any changes to `clCommandNDRangeKernelKHR`, you would automatically keep p to date with that.